HRIS.Services.Tests/Services/EmployeeSalaryDetailsServiceUnitTests.cs

```diff
@@ -0,0 +1,195 @@
+            _employeeSalaryDetailsMock = new Mock<EmployeeSalaryDetails>();
+            _employeeSalaryDetailsServiceMock = new Mock<IEmployeeSalarayDetailsService>();
+            _employeeSalaryDetailsService = new EmployeeSalaryDetailsService(_unitOfWorkMock.Object, _errorLoggingServiceMock.Object);
+            _employeeTypeServiceMock = new Mock<IEmployeeTypeService>();
+        }
+
+        int employeeId = 1;
+        static Employee testEmployee = new Employee(EmployeeTestData.EmployeeDto, EmployeeTypeTestData.DeveloperType);
+
+        [Fact]
+        public async Task GetEmployeeSalaryPass()
+        {
+            double salary = 25000.00;
+
+            var mockEmployeeDbSet = new List<Employee> { testEmployee }.AsQueryable().BuildMockDbSet();
+            _unitOfWorkMock.Setup(m => m.Employee.Get(It.IsAny<Expression<Func<Employee, bool>>>()))
+                        .Returns(mockEmployeeDbSet.Object);
+
+            var employeeSalaryDetails = new EmployeeSalaryDetails
+            {
+                EmployeeId = employeeId,
+                Salary = salary
+            };
+            var mockEmployeeSalaryDetailsDbSet = new List<EmployeeSalaryDetails> { employeeSalaryDetails }.AsQueryable().BuildMockDbSet();
+            _unitOfWorkMock.Setup(m => m.EmployeeSalaryDetails.Get(It.IsAny<Expression<Func<EmployeeSalaryDetails, bool>>>()))
+                           .Returns(mockEmployeeSalaryDetailsDbSet.Object);
+
+            var result = await _employeeSalaryDetailsService.GetEmployeeSalary(employeeId);
+
+            Assert.NotNull(result);
+            Assert.Equal(salary, result.Salary);
+        }
+
+        [Fact]
+        public async Task GetAllEmployeeSalariesPass()
+        {
+            var employeeSalaries = new List<EmployeeSalaryDetails>
+            {
+                new EmployeeSalaryDetails(EmployeeSalaryDetailsTestData.EmployeeSalaryTest1),
+                new EmployeeSalaryDetails(EmployeeSalaryDetailsTestData.EmployeeSalaryTest2)
+            };
+
+            var employeeSalariesDto = new List<EmployeeSalaryDetailsDto>
+            {
+                EmployeeSalaryDetailsTestData.EmployeeSalaryTest1,
+                EmployeeSalaryDetailsTestData.EmployeeSalaryTest2
+            };
+
+            var obj = _unitOfWorkMock
+            .Setup(m => m.EmployeeSalaryDetails.Get(It.IsAny<Expression<Func<EmployeeSalaryDetails, bool>>>()))
+            .Returns(employeeSalaries.AsQueryable().BuildMock());
+
+            _employeeSalaryDetailsServiceMock.Setup(r => r.GetAllEmployeeSalaries());
+
+            _unitOfWorkMock.Setup(x => x.EmployeeSalaryDetails.GetAll(null)).Returns(Task.FromResult(employeeSalariesDto));
+            var result = await _employeeSalaryDetailsService.GetAllEmployeeSalaries();
+
+            Assert.Equal(2, result.Count);
+            Assert.Equal(employeeSalariesDto, result);
+            Assert.NotNull(result);
+            Assert.IsType<List<EmployeeSalaryDetailsDto>>(result);
```

Comment:
```suggestion
            Assert.NotNull(result);
            Assert.Equal(2, result.Count);
            Assert.Equal(employeeSalariesDto, result);
            Assert.IsType<List<EmployeeSalaryDetailsDto>>(result);
```

There is no need to check if the value returned is null after you have already checked if the list has a size of two and if the dto is matching. Rather have the first check that it is not null then check if it contains the expected outcome

Reply:
Apologies, this was an oversight from my end. 

Your code suggestion and explanation does make sense. It doesn't make sense if we check for null after checking list size😅. I'll commit you suggestion😁